clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java

```diff
@@ -83,6 +86,10 @@ public OffsetFetchRequest build(short version) {
             }
 
             if (data.requireStable() && version < 7) {
+                if (throwOnStableFlagNotSupported) {
+                    throw new UnsupportedVersionException("Broker unexpectedly doesn't support requireStable flag " +
+                        "on version " + version);
+                }
```

Comment:
Could you double check if the UnsupportedVersionException would be thrown all the way up to `consumer.committed`, un-wrapped?

Also besides `consumer.committed` it would be thrown to other consumer calls, e.g. `consumer.poll` could be possibly sending offset fetch request as part of rebalance as well. We need to make sure all consumer APIs that may get this exception updated their javadocs.

Reply:
Addressed